src/Microsoft.AspNetCore.Routing/Matching/DfaMatcher.cs

```diff
@@ -229,5 +230,22 @@ public sealed override Task MatchAsync(HttpContext httpContext, IEndpointFeature
 
             return true;
         }
+
+        internal void ToGraph(TextWriter writer)
```

Comment:
What do you think of wrapping the graphvis stuff in `#if DEBUG`?

Reply:
Would it be possible to make this an external app or does it require internal access to the DfaMatcher data?

Reply:
I don't like the put things like behind debug, because the debug builds only happen in this repo. As a user you have no power to 'undebug' the bits we ship.

Do you have any cool ideas about how we could make this accessible to application authors? Ultimately I want us to ship some kind of experience for users to be able to dump the graph. I didn't try to solve all of the worlds problems in this PR.

Reply:
Yeah I coded this up last night and I've been stewing over it for about 18 hours.

Reply:
So I think what I want to do for now is turn this into a pubternal service that you can call given a data source. Then I'll add a sample to the routing sample that uses it in a static HTML page.